### ISSUE TITLE
Forecasting Stock Prices Using a Hybrid Deep Learning Model Integrating Attention Mechanism, Multi-Layer Perceptron, and Bidirectional Long-Short Term Memory Neural Network

### ISSUE THREAD

**Issue by jmathias001** [jmathias001/MathiasCapstone#11]
### Title
Forecasting Stock Prices Using a Hybrid Deep Learning Model Integrating Attention Mechanism, Multi-Layer Perceptron, and Bidirectional Long-Short Term Memory Neural Network
### URL
https://ieeexplore.ieee.org/abstract/document/9122554
### Summary 
Stock prices forecasting is a topic research in the fields of investment and national policy, which has been a challenging problem owing to the multi-noise, nonlinearity, high-frequency, and chaos of stocks. These characteristics of stocks impede most forecasting models from extracting valuable information from stocks data. Herein, a novel hybrid deep learning model integrating attention mechanism, multi-layer perceptron, and bidirectional long-short term memory neural network is proposed. First, the raw data including four types of datasets (historical prices of stocks, technical indicators of stocks closing prices, natural resources prices, and historical data of the Google index) are transformed into a knowledge base with reduced dimensions using principal component analysis. Subsequently, multi-layer perceptron is used for the fast transformation of feature space and rapid gradient descent, bidirectional long-short term memory neural network for extracting temporal features of stock time series data, and attention mechanism for making the neural network focus more on crucial temporal information by assigning higher weights. Finally, a comprehensive model evaluation method is used to compare the proposed model with seven related baseline models. After extensive experiments, the proposed model demonstrated its good forecasting performance.
### Key Points 
Time Series Analysis, Deep Learning, Long Short Term Memory
### Citation
Q. Chen, W. Zhang and Y. Lou, "Forecasting Stock Prices Using a Hybrid Deep Learning Model Integrating Attention Mechanism, Multi-Layer Perceptron, and Bidirectional Long-Short Term Memory Neural Network," in IEEE Access, vol. 8, pp. 117365-117376, 2020, doi: 10.1109/ACCESS.2020.3004284.
